### PR TITLE
Improved Settings and Styling Fixes

### PR DESCRIPTION
**Pull request by phschaad**
- [x] Various bugfixes
- [x] Improved backwards compatibility and added warnings if the dace version is not aligned with the specifically tested or recommended versions
- [x] Cleaned up styling and improved how some UI elements are shown
- [x] Slightly reduced deployment file size through removed dependencies
- [x] Adapted to improved settings UI / Interaction of SDFV

### CODE REVIEW

File: backend/dace_vscode/transformations.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def expand_library_node(json_in):
         if cfg_id is None:
             sdfg.expand_library_nodes()
         else:
-            if DACE_VERSION >= '0.16.0':
+            if DACE_VERSION >= '0.16.0' or not hasattr(sdfg, 'sdfg_list'):
```
Comment (tbennun) [inline backend/dace_vscode/transformations.py:45]: These are unsafe checks, ask instead if hasattr cfg_list. Also to be on the safe side with string comparisons you can make DACE_VERSION an int tuple once and check against (0, 16, 0)

File: scripts/generate_meta_dict.sh
Comment (tbennun) [inline scripts/generate_meta_dict.sh:1]: What’s this?

File: src/components/dace_interface.ts [hunk 2]
```diff
@@ -29,6 +29,9 @@ enum InteractionMode {
     APPLY,
 }
 
+const MIN_SAFE_VERSION = '0.16.0';
```
Comment (tbennun) [inline src/components/dace_interface.ts:32]: Same question as before. How does typescript do version checks in general?

File: src/webclients/components/sdfv/utils/helpers.ts [hunk 3]
```diff
@@ -97,44 +97,50 @@ export function findJsonSDFGElementByUUID(cfgList: CFGListType, uuid: string): [
...
+    }
+}
+
 /**
  * Perform an action for each element in an array given by their uuids.
  */
 export function doForAllUUIDs(
     uuids: string[], action: CallableFunction,
-    applyToScopeChildren: boolean = false
+    applyToScopeChildren: boolean = false, applyToNone: boolean = false
```
Comment (tbennun) [inline src/webclients/components/sdfv/utils/helpers.ts:128]: Apply to null/undefined? I know it’s from Python but it’s strange to see here. Also some doc would be nice.

**Review by tbennun — CHANGES_REQUESTED** [review]

File: scripts/generate_meta_dict.sh
Comment (phschaad) [reply scripts/generate_meta_dict.sh:1]: Made script executable

File: backend/dace_vscode/transformations.py (hunk 1 above)
Comment (phschaad) [reply backend/dace_vscode/transformations.py:45]: Why is the string comparison unsafe? Typescript / Javascript compares strings with regular numerical ordering, which means semver version strings can be compared as-is.

Comment (phschaad) [reply backend/dace_vscode/transformations.py:45]: With the cfg_list check the version can be dropped entirely.

File: src/webclients/components/sdfv/utils/helpers.ts (hunk 3 above)
Comment (phschaad) [reply src/webclients/components/sdfv/utils/helpers.ts]: Fixed

File: src/components/dace_interface.ts (hunk 2 above)
Comment (phschaad) [reply src/components/dace_interface.ts:33]: The semver package appears to be the production-grade industry standard - I adopted that too now

**Review by tbennun — APPROVED** [review]
Thanks for addressing the comments. LGTM

**Pull request merged**